sdks/cpp/src/agones/sdk.cc

```diff
@@ -47,84 +43,85 @@ bool SDK::Connect() {
   stable::agones::dev::sdk::Empty response;
 
-  return pimpl_->stub_->Ready(context, request, &response);
+  return pimpl_->stub_->Ready(&context, request, &response);
 }
 
 bool SDK::Health() {
   stable::agones::dev::sdk::Empty request;
   return pimpl_->health_->Write(request);
 }
 
-grpc::Status SDK::GameServer(stable::agones::dev::sdk::GameServer *response) {
-  grpc::ClientContext *context = new grpc::ClientContext();
-  context->set_deadline(gpr_time_add(gpr_now(GPR_CLOCK_REALTIME),
-                                     gpr_time_from_seconds(30, GPR_TIMESPAN)));
+grpc::Status SDK::GameServer(stable::agones::dev::sdk::GameServer* response) {
+  grpc::ClientContext context;
+  context.set_deadline(gpr_time_add(gpr_now(GPR_CLOCK_REALTIME),
+                                    gpr_time_from_seconds(30, GPR_TIMESPAN)));
   stable::agones::dev::sdk::Empty request;
 
-  return pimpl_->stub_->GetGameServer(context, request, response);
+  return pimpl_->stub_->GetGameServer(&context, request, response);
 }
 
 grpc::Status SDK::WatchGameServer(
-    const std::function<void(stable::agones::dev::sdk::GameServer)> &callback) {
-  grpc::ClientContext *context = new grpc::ClientContext();
+    const std::function<void(stable::agones::dev::sdk::GameServer)>& callback) {
+  grpc::ClientContext context;
   stable::agones::dev::sdk::Empty request;
   stable::agones::dev::sdk::GameServer gameServer;
 
   std::unique_ptr<grpc::ClientReader<stable::agones::dev::sdk::GameServer>>
-      reader = pimpl_->stub_->WatchGameServer(context, request);
+      reader = pimpl_->stub_->WatchGameServer(&context, request);
   while (reader->Read(&gameServer)) {
     callback(gameServer);
   }
   return reader->Finish();
 }
 
 grpc::Status SDK::Shutdown() {
-  grpc::ClientContext *context = new grpc::ClientContext();
-  context->set_deadline(gpr_time_add(gpr_now(GPR_CLOCK_REALTIME),
-                                     gpr_time_from_seconds(30, GPR_TIMESPAN)));
+  grpc::ClientContext context;
+  context.set_deadline(gpr_time_add(gpr_now(GPR_CLOCK_REALTIME),
+                                    gpr_time_from_seconds(30, GPR_TIMESPAN)));
   stable::agones::dev::sdk::Empty request;
   stable::agones::dev::sdk::Empty response;
 
-  return pimpl_->stub_->Shutdown(context, request, &response);
+  return pimpl_->stub_->Shutdown(&context, request, &response);
 }
 
 grpc::Status SDK::SetLabel(std::string key, std::string value) {
-  grpc::ClientContext *context = new grpc::ClientContext();
-  context->set_deadline(gpr_time_add(gpr_now(GPR_CLOCK_REALTIME),
-                                     gpr_time_from_seconds(30, GPR_TIMESPAN)));
+  grpc::ClientContext context;
+  context.set_deadline(gpr_time_add(gpr_now(GPR_CLOCK_REALTIME),
+                                    gpr_time_from_seconds(30, GPR_TIMESPAN)));
 
   stable::agones::dev::sdk::KeyValue request;
-  request.set_key(key);
-  request.set_value(value);
+  request.set_key(std::move(key));
```

Comment:
std::move is redundant here and in next calls of `set_key` and `set_value` because it accepts argument via const reference.

Reply:
Since C++11, protos also have setters that take rvalues to support efficient moves. To quote from [this doc](https://developers.google.com/protocol-buffers/docs/reference/cpp-generated):

> `void set_foo(string&& value)` (C++11 and beyond): Sets the value of the field, moving from the passed string. After calling this, foo() will return a copy of value.

So I think we want to do one of the following:

A: Keep these moves, or ...
B: Change the function's argument types to be `const std::string&` and not move. 

I recommend (A).